quipucords/api/scan/serializer.py

```diff
@@ -16,31 +16,76 @@
 from rest_framework.serializers import (PrimaryKeyRelatedField,
                                         ValidationError,
                                         IntegerField,
-                                        CharField)
+                                        CharField,
+                                        BooleanField)
 from api.models import (Source,
                         ScanTask,
                         Scan,
-                        ScanOptions)
+                        ScanOptions,
+                        ExtendedProductSearchOptions)
 import api.messages as messages
 from api.common.serializer import (NotEmptySerializer,
                                    ValidStringChoiceField,
                                    CustomJSONField)
 from api.scantasks.serializer import SourceField
 from api.common.util import check_for_existing_name
 
+try:
+    json_exception_class = json.decoder.JSONDecodeError
+except AttributeError:
+    json_exception_class = ValueError
+
+
+class ExtendedProductSearchOptionsSerializer(NotEmptySerializer):
+    """The extended production search options of a scan."""
+
+    jboss_eap = BooleanField(required=False)
+    jboss_fuse = BooleanField(required=False)
+    jboss_brms = BooleanField(required=False)
+    search_directories = CustomJSONField(required=False)
+
+    class Meta:
+        """Metadata for serializer."""
+
+        model = ExtendedProductSearchOptions
+        fields = ['jboss_eap',
+                  'jboss_fuse',
+                  'jboss_brms',
+                  'search_directories']
+
+    @staticmethod
+    def validate_search_directories(search_directories):
+        """Validate search directories."""
+        try:
+            search_directories_list = json.loads(search_directories)
+            if not isinstance(search_directories_list, list):
+                raise ValidationError(
+                    _(messages.SCAN_OPTIONS_EXTENDED_SEARCH_DIR_NOT_LIST))
+            for directory in search_directories_list:
+                if not isinstance(directory, str):
+                    raise ValidationError(
+                        _(messages.SCAN_OPTIONS_EXTENDED_SEARCH_DIR_NOT_LIST))
```

Comment:
Worth adding a check like the following:
https://github.com/quipucords/rho/blob/master/rho/utilities.py#L578

Makes sure what was given was at minimum a "good" path structure.

Reply:
Done.